packages/cli/src/ui/App.tsx

```diff
@@ -465,7 +493,21 @@ const App = ({ config, settings, startupWarnings = [], version }: AppProps) => {
         handleSlashCommand(newValue ? '/mcp desc' : '/mcp nodesc');
       }
     } else if (key.ctrl && (input === 'c' || input === 'C')) {
-      handleExit(ctrlCPressedOnce, setCtrlCPressedOnce, ctrlCTimerRef);
+      // Check if input buffer has content
```

Comment:
why is this an improvement over the previous logic that handled this within InputPrompt?